src/hypothesis/settings.py

```diff
@@ -198,6 +201,10 @@ def define_setting(cls, name, description, default, options=None):
         all_settings[name] = Setting(
             name, description.strip(), default, options)
         setattr(cls, name, SettingsProperty(name))
+        if cls.default:
+            setattr(cls.default, name, default)
+        for profile in cls._profiles.values():
```

Comment:
I'm very confused what's going on here, but this PR is the first thing I do when drinking my coffee so I also would not be surprised if I were being dense. :-)

Why is the setattr(profile, name, default) not clobbering this? There must be an attribute lookup happening before the assignment I guess, but relying on that seems very suspect.

OTOH with the test there I don't mind too much. The settings system is due for a significant rework and I'm happy to get this merged and do the internals cleanup myself.